tests/tests_bind_high_availability.yml

```diff
@@ -0,0 +1,36 @@
+---
+- hosts: all
+
+  vars:
+    test_device: /tmp/.nbde_client_dev_test
+    test_pass: test-password-here
+
+    nbde_client_bindings:
+      - devices:
+          - path: "{{ test_device }}"
+            pass: "{{ test_pass }}"
+        pin:
+          servers:
+            - public.nbde.cc
+            - public2.nbde.cc
+
+  tasks:
+    - name: Set up test device
+      include: create_dummy_device.yml
+
+    - name: Use nbde_client role
+      include_role:
+        name: linux-system-roles.nbde_client
+
+    - name: Attempt to unlock device
+      include: verify_unlock_device.yml
+
+    - name: Make sure the attempt to unlock succeeded
+      assert:
+        that:
+          - not nbde_unlock.failed
+          - not nbde_close.failed
+
+    - name: Clean up
```

Comment:
The cleanup won't be executed if `name: Make sure the attempt to unlock succeeded` asserts

Reply:
I will try to use the approach you suggested with the block/rescue/always. Hopefully I can get rid of the `ignore_errors: yes` as well in the verification tasks

Reply:
> Hopefully I can get rid of the `ignore_errors: yes` as well in the verification tasks

It depends - you may actually want to run all of the verification tasks, and report all of the errors, rather than just erroring out on the first one.  It's up to you to decide what would be preferable for certain tests.  Here is an approach I took for kernel_settings - https://github.com/linux-system-roles/kernel_settings/pull/6/commits/fa71b5e9140fa2889b1336a0932b98ab105d75e6#diff-67385c2b6739d65912ec02759af26815R159

Reply:
Okay, I am keeping with `ignore_errors` + `register` and checking the results. I am also using block/always in the tests, so that cleanup is always executed.